app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java

```diff
@@ -104,6 +112,7 @@ private void initRecyclerView() {
         }
 
         rvContributionsList.setAdapter(adapter);
+        adapter.setContributions(contributions);
```

Comment:
This is essentially a no-op, can be deleted

Reply:
set contributions is not always called, with the existing-current implementation, it is only called the first time or after the threshold for the fetch-contributions timestamp has passed, not maintaining a local copy with the fragment would leave it's adapter with empty list of contributions on recreations and that is why this is nescessary

Reply:
This is an empty list, it will always be an empty list at this moment in time. You do not persist this list in the bundle. It is a no-op.

Reply:
It will not always be empty, setContributions adds value to this list

Reply:
>it will always be an empty list at this moment in time.

There is one scenario, if the fragment doesn't actually get destroyed.
![https___cdn-images-1 medium com_max_2000_0_YWCr1M1pwWfvCzvZ](https://user-images.githubusercontent.com/3358282/75972538-7e62d600-5ecb-11ea-9908-42b69b1fa578.png)

How is the adapter losing the data but this list isn't? maybe that problem should be addressed